### PR TITLE
TASK-60: Authentication for Subscription

### LINKED ISSUE #215: TASK-60: Authentication for Subscription
**Task Description**

As a developer, I want to add Authentication to the post subscription so that only users with the valid authToken can fetch the posts.

**Acceptance Criteria**

- [ ] Only validated users can fetch posts 
- [ ] Have a testable front-end subscription working 
- [ ] Documentation present where necessary
- [ ] Unit and UI tests are implemented for every new component and function
- [ ] The application should still work with seeded data

### PR DESCRIPTION
**Pull request by bit172**
This PR closes #215.

**Steps to test**
1. Start the application.
2. Log in to MullApp.
3. Navigate to `/messages/announcements`.
4. Open your favorite GraphQL playground.
5. Send a post using the `post` mutation.
 ```graphql
    mutation Mutation($postPost: CreatePostInput!) {
      post(post: $postPost) {
        id
      }
    }
``` 
Variables:
```JSON
{
  "postPost": {
    "message": "Hello there!",
    "createdTime": "2021-02-28T16:11:00.000Z",
    "user": {
      "id": 1
    }
  }
}
```
(Don't forget the authorization header)
6. You should see the post as a JSON in the browser.

### CODE REVIEW

**Review by JimLau49 — APPROVED** [review]
Good work on implementing the authenticated websocket link! subscriptions working seamlessly when authenticated properly
image

File: apps/mull-ui/src/environments/environment.ts [hunk 1]
```diff
@@ -4,4 +4,5 @@
 export const environment = {
   production: false,
   backendUrl: 'http://localhost:3333',
+  backendWsUrl: 'ws://localhost:3333',
```
Comment (RGPosadas) [inline apps/mull-ui/src/environments/environment.ts:7]: Pleas don't forget to add prod equivalent

File: apps/mull-ui/src/app/pages/messages/messages-page.tsx [hunk 2]
```diff
@@ -1,17 +1,27 @@
...
 export interface MessagePageProps {
   children: React.ReactNode;
 }
+
 /**
  * Adds a sub-nav-header and sub-navigation to the top of the page, above the children
  * Children are rendered in between and then adds a sticky chat-input at the bottom of the page.
  */
 export const MessagesPage = ({ children }: MessagePageProps) => {
+  const { data, loading } = usePostAddedSubscription();
```
Comment (RGPosadas) [inline apps/mull-ui/src/app/pages/messages/messages-page.tsx:15]: Don't forget to update the snapshot and add MockedProvider

**Review by RGPosadas — APPROVED** [review]
Approved, great work 🚀 ! Our subscription is no longer naked 😆 

Had two small comments, however they are not blockers. Feel free to push without re-requesting my review 👍

File: apps/mull-ui/src/environments/environment.ts (hunk 1 above)
Comment (bit172) [reply apps/mull-ui/src/environments/environment.ts:7]: Where should that be since I can't seem to find one for the backend Url in the frontend app?

Comment (RGPosadas) [reply apps/mull-ui/src/environments/environment.ts:7]: It seems that with our deployment, WebSockets are handled by NGINX out of the box. So we can leave the prod equivalent with an empty string for now.

Ref: https://kubernetes.github.io/ingress-nginx/user-guide/miscellaneous/#websockets

**Review by leojrsilao — APPROVED** [review]
LGTM! 🚀 

- Can add posts when logged in with authorized users. 
- Linter and tests pass

**Pull request merged**